x/evm/keeper/integration_test.go

```diff
@@ -499,7 +511,9 @@ var _ = Describe("Handling a MsgEthereumTx message", Label("EVM"), Ordered, func
 		res, err := s.factory.ExecuteContractCall(callSigner, totalSupplyTxArgs, totalSupplyArgs)
 		if callParams.ExpFail {
 			Expect(err).NotTo(BeNil())
+			// TODO: Weird another error message is returned instead of the expected one
```

Comment:
it used to fail on gas estimation, now if we define the gas limit it fails on the ante handler.
we have the same check on two different places.

Reply:
Hmm that makes sense but we have to define a gas limit here otherwise when we scale it to 6 decimals it fees become 0 and it fails.